app/controllers/general_shopping_list_controller.rb

```diff
@@ -0,0 +1,6 @@
+class GeneralShoppingListController < ApplicationController
+  def index
+    @users = User.all.includes(:foods)
+    @foods = Food.where.not(user_id: current_user.id).includes(:user)
```

Comment:
- [ ] According to the [project requirements](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/buisness_requirements.md) the general shopping list should show the list of food that is missing for all recipes of the logged-in user.
- [ ]  This means you should  compare the `current_user`  foods with all the foods in the recipes of that user.
- [ ]  I suggest a code like this in order to meet the project requirements :
    ```suggestion
  class GeneralShoppingListController < ApplicationController
  def index
    @recipes = current_user.recipes.includes(:foods)
    @general_food_list = current_user.foods

    @missing_food_list = find_missing_food_list
  end

  private

  def find_missing_food_list
    missing_foods = []

    @recipes.each do |recipe|
      recipe.foods.each do |food|
        missing_foods << food unless @general_food_list.include?(food)
      end
    end

    missing_foods
  end
end
```